spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/support/JdbcQueryMethodUnitTests.java

```diff
@@ -57,15 +65,80 @@ public void returnsSpecifiedRowMapperClass() throws NoSuchMethodException {
 
 		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
 
-		JdbcQueryMethod queryMethod = new JdbcQueryMethod(JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethod"),
-				metadata, mock(ProjectionFactory.class));
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethod"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
 
 		assertThat(queryMethod.getRowMapperClass()).isEqualTo(CustomRowMapper.class);
 	}
 
-	@Query(value = DUMMY_SELECT, rowMapperClass = CustomRowMapper.class)
-	private void queryMethod() {}
+	@Test // DATAJDBC-234
+	public void returnsSqlStatementName() throws NoSuchMethodException {
+
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethodName"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+
+		assertThat(queryMethod.getAnnotatedQuery()).isEqualTo(DUMMY_SELECT_VALUE);
+	}
+	@Test // DATAJDBC-234
+	public void returnsSqlStatementNameAndValue() throws NoSuchMethodException {
+		
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+		
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+		
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+		
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethodNameAndValue"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+		
+		assertThat(queryMethod.getAnnotatedQuery()).isEqualTo(DUMMY_SELECT_NAME_VALUE);
+	}
+
+	@Test // DATAJDBC-234
+	public void returnsSpecifiedRowMapperClassName() throws NoSuchMethodException {
+
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethodName"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+
+		assertThat(queryMethod.getRowMapperClass()).isEqualTo(CustomRowMapper.class);
+	}
+
```

Comment:
There should be another test without an `@Query` annotation just based on the default naming.

Reply:
Done. Add a method without query. the methodQuery returns null. like you want . Thanks